DomainManagement/functions/AccessRule/Invoke-DMAccessRule.ps1

```diff
@@ -137,14 +148,25 @@
 							try {
 								if (-not $changeEntry.Configuration.ObjectType) { throw "Unknown ObjectType! Unable to translate $($changeEntry.Configuration.ObjectTypeName). Validate the configuration and ensure pending schema updates (e.g. Exchange, Skype, etc.) have been applied." }
 								if (-not $changeEntry.Configuration.InheritedObjectType) { throw "Unknown InheritedObjectType! Unable to translate $($changeEntry.Configuration.InheritedObjectTypeName). Validate the configuration and ensure pending schema updates (e.g. Exchange, Skype, etc.) have been applied." }
-								$accessRule = [System.DirectoryServices.ActiveDirectoryAccessRule]::new((Convert-Principal @parameters -Name $changeEntry.Configuration.IdentityReference), $changeEntry.Configuration.ActiveDirectoryRights, $changeEntry.Configuration.AccessControlType, $changeEntry.Configuration.ObjectType, $changeEntry.Configuration.InheritanceType, $changeEntry.Configuration.InheritedObjectType)
+								$accessRule = [System.DirectoryServices.ActiveDirectoryAccessRule]::new(
+                                    (Convert-Principal @parameters -Name $changeEntry.Configuration.IdentityReference),
+                                    $changeEntry.Configuration.ActiveDirectoryRights,
+                                    $changeEntry.Configuration.AccessControlType,
+                                    $changeEntry.Configuration.ObjectType,
+                                    $changeEntry.Configuration.InheritanceType,
+                                    $changeEntry.Configuration.InheritedObjectType
+                                )
 							}
 							catch {
 								$failedCount = $failedCount + 1
 								Stop-PSFFunction -String 'Invoke-DMAccessRule.AccessRule.Creation.Failed' -StringValues $testItem.Identity, $changeEntry.Configuration.IdentityReference -EnableException $EnableException -Target $changeEntry -Continue -ErrorRecord $_
 							}
-							$null = $aclObject.AddAccessRule($accessRule)
-							#TODO: Validation and remediation of success. Adding can succeed but not do anything, when accessrules are redundant. Potentially flag it for full replacement?
+                            $null = $aclObject.AddAccessRule($accessRule)
+                            # Validation and remediation of success. Adding can succeed but not do anything, when accessrules are redundant. Potentially flag it for full replacement?
+                            if(-not ($aclObject.Access | Where-Object { $_ -in $accessRule })) {
+                                $failedCount = $failedCount + 1
+                                Write-PSFMessage -Level Warning -String 'Invoke-DMAccessRule.AccessRule.Creation.NotApplied' -StringValues $testItem.Identity, $changeEntry.Configuration.IdentityReference -Target $changeEntry
+                            }
```

Comment:
Was there a particular problem you added those for or was that more of a "Oh. there's a TODO marker there, lets add that while I'm on it"?

Reply:
Actually, I was digging around with the AccessRules and did some testing. As far as I remember -and it was a while ago, so my mind is a bit blurry on the exact situation- I did consider that as a bug.
ADMF did not throw information on NoFix-Items and on edge-case-scenarios.
Something like a ACE is not present on the object but defined on schema default ACL. Something like this, but as said, I did not remember exactly.
...and in the end... in fact it was marked as ToDo and did make sense to me.